### DISCUSSION TITLE
I was not able to get the I2StoBuilt-in DAC sample to work.

### DISCUSSION THREAD

**Discussion by riraosan** [pschatzmann/arduino-audio-tools#83] [Q&A] [answered]
streams-analog-I2S
streams-generator-i2s
streams-memory_mp3-analog
streams-sd_mp3-i2s

I tried to run the I2S related samples, but they didn't work using the samples as they were. (Doesn't the sample work with the latest library?)

When I deciphered the library, I found that there were many parameters that I had to set that were not written in the sample code.
In the end, however, I was able to get audio out of the built-in DAC, but the sound was too distorted to use.

The audio output was more distorted than the 25 or 26 pin output of the M5 ATOM Lite, and even a filter did not improve the sound. I checked the operation by setting some parameters at random, but it did not improve.

So, I gave up. My skills were not good enough.

I appreciate your library. 👍
However, I think you should improve the explanation of how to check and configure the samples.

PS I have confirmed that your library related to PWM works.

**Comment by pschatzmann** [comment, marked as answer]
Which external DAC did you try to use ?
All I2S examples require an external DAC. Only the analog examples are using the built in DAC of the ESP32, but the quality is usually not good because there is a lot of noise from the power supply.
